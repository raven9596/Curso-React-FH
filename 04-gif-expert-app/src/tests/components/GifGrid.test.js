import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el <GifGrid />', () => {

  const category = 'One Punch';
  
  test('debe de mostrarse correctamente', () => {

    useFetchGifs.mockReturnValue({
			data: [],
      loading: true,
      
		});
    
    const wrapper = shallow(<GifGrid category={ category }/>);
    expect( wrapper ).toMatchSnapshot();
    
  });

  test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

    const gifs = [{
      id: 'ABC',
      url: 'https://localhost/cualquier/cosa.jpg',
      title: 'Cualquier cosa',
    }]

    useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		});
    
    const wrapper = shallow(<GifGrid category={category} />);

    expect( wrapper ).toMatchSnapshot();
    
  });
  
});