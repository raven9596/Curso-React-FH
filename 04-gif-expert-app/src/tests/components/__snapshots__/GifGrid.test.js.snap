// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flash animate__delay-2s"
  >
    One Punch
  </h3>
  <p>
     loading 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
